src/app/infuse/infuse.component.js

```diff
@@ -37,37 +37,32 @@ function InfuseCtrl($scope, dimStoreService, D2StoresService, dimDefinitions, D2
   angular.extend(vm, {
     getAllItems: true,
     showLockedItems: false,
+    source: null,
     target: null,
-    exotic: false,
     infused: 0,
-    infusable: [],
+    sourceItems: [],
+    targetItems: [],
     transferInProgress: false,
 
     $onInit: function() {
       // Set the source and reset the targets
       vm.infused = 0;
       vm.target = null;
-      vm.exotic = vm.source.tier === 'Exotic';
-      vm.stat = vm.source.primStat.stat;
-      if (vm.source.bucket.sort === 'General') {
-        vm.wildcardMaterialCost = 2;
-        vm.wildcardMaterialHash = 937555249;
-      } else if (vm.stat.statIdentifier === 'STAT_DAMAGE') {
-        vm.wildcardMaterialCost = 10;
-        vm.wildcardMaterialHash = 1898539128;
-      } else {
-        vm.wildcardMaterialCost = 10;
-        vm.wildcardMaterialHash = 1542293174;
-      }
 
       vm.getItems();
     },
 
-    selectItem: function(item, e) {
+    selectItem: function(item, isTarget, e) {
```

Comment:
Functions that take a boolean parameter are often hard to understand at the call site (for example, in the template, `selectItem(item, true, $event)` doesn't give me a lot to go on. It might be nice to give named functions that call into this one called `selectSourceItem` or `selectTargetItem`, for example.

Reply:
Another suggestion might be to call it using the intended `target` and `source` right in the parameters if possible. So the function signature might look like `selectItem: function(targetItem, sourceItem, e)` instead.